src/lib/api/githubClient.ts

```diff
@@ -0,0 +1,64 @@
+import { useQuery } from '@tanstack/react-query'
+import { AVERAGE_BLOCKTIME } from '../constants'
+
+interface GitHubContent {
+  type: string
+  encoding: string
+  size: number
+  name: string
+  path: string
+  content: string
+  sha: string
+  url: string
+  git_url: string
+  html_url: string
+}
+
+export const fetchFileContent = async (
+  owner: string,
+  repo: string,
+  path: string,
+  ref?: string,
+  token?: string,
+): Promise<string> => {
+  const url = `https://api.github.com/repos/${owner}/${repo}/contents/${path}${ref ? `?ref=${ref}` : ''}`
```

Comment:
Any particular reason for using the API rather than `raw.githubusercontent.com` endpoint? From what I can see the API have a [rate limit of 60 requests per hour](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users).

Reply:
not really, didn't know about the limit, though it makes sense. I'll change it